python/ray/serve/router.py

```diff
@@ -263,24 +272,20 @@ async def assign_request(
         if not self._pending_endpoint_registered.is_set():
             # This can happen when the router is created but the endpoint
             # information hasn't been retrieved via long-poll yet.
-            try:
-                await asyncio.wait_for(
-                    self._pending_endpoint_registered.wait(),
-                    timeout=5,
-                )
-            except asyncio.TimeoutError:
+            # Block the whole world wait, this should be very quick.
+            is_set = self._pending_endpoint_registered.wait(timeout=5)
+            if not is_set:
                 raise RayServeException(
                     f"Endpoint {endpoint} doesn't exist after 5s timeout. "
                     "Marking the query failed.")
 
         chosen_backend, *shadow_backends = self.endpoint_policy.assign(query)
 
-        result_ref = await self._get_or_create_replica_set(
-            chosen_backend).assign_replica(query)
+        future_ref = self.backend_replicas[chosen_backend].assign_replica(
```

Comment:
nit, but I'd call this `ref_future` because it's a future to an object_ref, not an object_ref to a future